src/EventSubscriber/LinkTypePlusEventSubscriber.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Drupal\commerce_gmo_linktypeplus\EventSubscriber;
+
+use Drupal\commerce_gmo_linktypeplus\Event\LinkTypePlusEvent;
+use Drupal\commerce_order\Entity\Order;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Orderstatus change event subscriber.
+ */
+class LinkTypePlusEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Logger Factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $loggerFactory;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $loggerFactory
+   *   Logger .
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, LoggerChannelFactoryInterface $loggerFactory) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->loggerFactory = $loggerFactory->get('linktype_event_subscriber');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      'Credit' => 'onlinkTypePaymentEvent',
+      'PayPay' => 'onlinkTypePaymentEvent',
+      'CVS' => 'onlinkTypePaymentEvent',
+      'PayEasy' => 'onlinkTypePaymentEvent',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function onlinkTypePaymentEvent(LinkTypePlusEvent $event) {
+    $paymentMethod = $event->getPaymentMethod();
+    // Get the payment status and update the payment in Drupal.
+    $orderId = $event->getOrderId();
+     //remove version from orderId
+    $orderStripId = explode("-", $orderId)[0];
+    $status = $event->getTransitionState();
+    $drupalStatus = $this->webhookStatusMapper($status);
+    $remoteId = $event->getRemoteId();
+    $this->loggerFactory->notice("$paymentMethod payment event has been subscribed");
+    $this->loggerFactory->notice("$orderStripId \n  $status \n $remoteId");
+    if (!empty($orderStripId) && !empty($drupalStatus) && !empty($remoteId)) {
+      if ($this->updatePaymentStatus($orderStripId, $drupalStatus, $remoteId)) {
+        $this->loggerFactory->notice('Status has been updated');
+        return TRUE;
+      }
+    }
+  }
+
+
+  /**
+   * Update the status in Drupal.
+   *
+   * @param array $data
+   *   The GMO api response data.
+   */
+  public function updatePaymentStatus($order_id, $status, $remote_id) {
```

Comment:
Feature Request : This fn we are using in completeUrl also. If the funtionality doesnt differ, we can have a common method instead of repeting the code